assets/proposals/nnnn-audio-io-manager/SDLAudioIOManager.m

```diff
@@ -0,0 +1,746 @@
+    for (SDLAudioPassThruCapabilities *item in self.sdlManager.registerResponse.audioPassThruCapabilities) {
+        if (![item.audioType isEqualToEnum:SDLAudioTypePCM]) continue;
+        if (![item.bitsPerSample isEqualToEnum:SDLBitsPerSample16Bit]) continue;
+        if (![item.samplingRate isEqualToEnum:SDLSamplingRate16KHZ]) continue;
+        
+        audioOptions = item;
+        break;
+    }
+    
+    // split the text into multiple lines.
+    NSArray<NSString *> *lines = [self.inputStreamText componentsSeparatedByCharactersInSet:[NSCharacterSet newlineCharacterSet]];
+    
+    self.inputStreamOptions = audioOptions;
+    self.inputStreamAmplifierFactor = 0;
+    
+    // create a request we will be sending
+    SDLPerformAudioPassThru *performAudioInput = [[SDLPerformAudioPassThru alloc] init];
+    performAudioInput.audioType = audioOptions.audioType;
+    performAudioInput.bitsPerSample = audioOptions.bitsPerSample;
+    performAudioInput.samplingRate = audioOptions.samplingRate;
+    performAudioInput.initialPrompt = [NSArray arrayWithObject:self.inputStreamPrompt];
+    performAudioInput.maxDuration = @1000000;
+    performAudioInput.audioPassThruDisplayText1 = lines.count > 0 ? lines[0] : nil;
+    performAudioInput.audioPassThruDisplayText2 = lines.count > 1 ? lines[1] : nil;
+    performAudioInput.muteAudio = @YES;
+    
+    __weak SDLAudioIOManager * weakSelf = self;
+    
+    // this is the important area... handle the microphone audio data
+    performAudioInput.audioDataHandler = ^(NSData * _Nullable audioData) {
+        __strong SDLAudioIOManager * strongSelf = weakSelf;
+        if (strongSelf == nil || audioData == nil) {
+            return;
+        }
+        
+        [strongSelf sdl_initiateAbortInputStreamTimer];
+        
+        SDLLogV(@"Receiving audio data (length %lu)", (unsigned long)audioData.length);
+        
+        dispatch_async(strongSelf.dispatchQueue, ^{
+        [strongSelf sdl_handleInputStreamData:audioData];
+        });
+    };
+    
+    [self sdl_initiateAbortInputStreamTimer];
+    
+    // send the request out to the head unit
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+    SDLLogV(@"Sending request %@", [performAudioInput serializeAsDictionary:0]);
+    [self.sdlManager sendRequest:performAudioInput withResponseHandler:^(__kindof SDLRPCRequest * _Nullable request, __kindof SDLRPCResponse * _Nullable response, NSError * _Nullable error) {
+        SDLLogV(@"Response received %@", [response serializeAsDictionary:0]);
+#pragma clang diagnostic pop
+        
+        if (error) {
+            SDLLogW(@"Audio IO manager: Error in response %@", error);
+        }
+        
+        __strong SDLAudioIOManager * strongSelf = weakSelf;
+        if (strongSelf == nil) {
+            return;
+        }
+        
+        //TODO: Delayed Selector. Expectation fulfilled.
```

Comment:
Should this `TODO` be completed?

Reply:
Sorry. This is a left over to cancel a previously delayed selector. The expectation of receiving a response was fulfilled. I can remove this comment.